client/blocks/editor-checkout-modal/index.tsx

```diff
@@ -23,74 +28,72 @@ import './style.scss';
 
 const wpcom = wp.undocumented();
 
-type Site = {
-	ID: number;
-	slug: string;
-};
-
-export interface CartData {
-	products: Array< {
-		product_id: number;
-		product_slug: string;
-	} >;
+async function getCart( site: SiteData, cartData: CartData ) {
+	try {
+		return await wpcom.setCart( site.ID, cartData );
+	} catch {
+		return;
+	}
 }
 
-type Props = {
-	site: Site;
-	cartData: CartData;
-	onClose: () => void;
-	isOpen: boolean;
-};
-
-class EditorCheckoutModal extends Component< Props > {
-	static defaultProps = {
-		isOpen: false,
-		onClose: () => null,
-		cartData: {},
-	};
-
-	async getCart() {
-		// Important: If getCart or cartData is empty, it will redirect to the plans page in customer home.
-		const { site, cartData } = this.props;
+function fetchStripeConfigurationWpcom( args: Record< string, unknown > ) {
+	return fetchStripeConfiguration( args, wpcom );
+}
 
-		try {
-			return await wpcom.setCart( site.ID, cartData );
-		} catch {
-			return;
-		}
-	}
+const EditorCheckoutModal = ( props: Props ) => {
+	const { site, isOpen, onClose, cartData } = props;
+	const hasEmptyCart = ! cartData.products || cartData.products.length < 1;
 
-	render() {
-		const { site, isOpen, onClose, cartData } = this.props;
+	const user = userFactory();
+	const isLoggedOutCart = ! user?.get();
+	const waitForOtherCartUpdates = false;
```

Comment:
I am not too sure what this value is for, or if we can safely assume it's value is going to be false.

It's also used in [this file](https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/checkout/checkout-system-decider.js) to reference but could use another opinion on this.

Reply:
I think we'll have to ask @sirbrillig for this one.

Reply:
There are two situations where the ShoppingCartProvider needs to wait for the old CartStore in new checkout. I'm not sure if either of them apply to your use-case.

1. If some other page has added a product to the cart and then redirected to checkout without waiting for the cart transaction to complete. This creates a race condition between the two cart managers, since ShoppingCartProvider might query the cart before the change request completes, causing it to get the wrong version of the cart. This should really be fixed on the other end by waiting before the page redirects, but that requires finding everywhere that might modify the cart in calypso and I haven't had the time yet. So `waitForOtherCartUpdates` is true if the CartStore is pending updates.
2. The Registrationless checkout flow uses localStorage to keep a list of products for the cart which are loaded by the CartStore during signup. We need to make sure that when the ShoppingCartProvider loads the cart, that those changes have completed and that we pull them in from the CartStore properly. Ultimately, we should be solving this by making the ShoppingCartProvider capable of dealing with localStorage instead of the CartStore, but I haven't had time for that yet.